src/librawspeed/decoders/TiffBasedDecoder.h

```diff
@@ -0,0 +1,40 @@
+/*
+    RawSpeed - RAW file decoder.
+
+    Copyright (C) 2017 Axel Waggershauser
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#pragma once
+
+#include "decoders/RawDecoder.h" // for RawDecoder
+#include "io/FileMap.h"          // for FileMap
+#include "tiff/TiffIFD.h"        // for TiffRootIFDOwner
+
+namespace RawSpeed {
+
+class TiffBasedDecoder : public RawDecoder
```

Comment:
That is fine with me. To explicitly spell it out: `AbstractTiffEncoder`?

My 'problem' with this prefix is just that I think the prefix is justified when you want to explicitly state that this class is abstracting away some behavior. While that is true for the suggested class, it is not really because of the Tiff stuff but the RawEncoder stuff. The tiff-part does not add any abstract methods. But that is really a minute detail and subjective as well. I'm totall fine with `AbstractTiffEncoder`.